### PR TITLE
fixed AudioFileProcessor reverse bug

### PR DESCRIPTION
**Pull request by szeli1**
#6708 fixed

The loop knob will not move randomly when the sample is reversed even when the loop knob is at the end.
The problem was caused by the start knob clamping the loop knob's value because it was moved earlier.

### CODE REVIEW

**Review by zonkmachine — APPROVED** [review]
Tests fine using the floating point debug extensions, `-DWANT_DEBUG_FPE`.
In my experience the afp is one of the more confusing places of the code so I'm happy to see that I actually get what the proposed change is doing.

PS. I would give the comment a space after the comment mark:

```cpp
// update knobs in the right order
// to avoid them clamping each other

```

**Comment by szeli1** [comment]
Thank you (this is my first pr approved ever), at first I thought the comments were unnecessary (I don't see a lot in the code base), but changed them anyways.
Is there some guideline about when should I add comments?

**Comment by zonkmachine** [comment]
> Is there some guideline about when should I add comments?

I just thought it looked better that way.

File: plugins/AudioFileProcessor/AudioFileProcessor.cpp [hunk 1]
```diff
@@ -1160,14 +1160,37 @@ void AudioFileProcessorWaveView::slideSampleByFrames( f_cnt_t _frames )
...
+			m_startKnob->slideBy(v, false);
+		}
+		if (m_loopKnob)
+		{
+			m_loopKnob->slideBy(v, false);
+		}
+		if (m_endKnob)
+		{
+			m_endKnob->slideBy(v, false);
+		}
```
Comment (sakertooth) [inline plugins/AudioFileProcessor/AudioFileProcessor.cpp:1178]:
```suggestion
		m_startKnob->slideBy(v, false);
		m_loopKnob->slideBy(v, false);
		m_endKnob->slideBy(v, false);
```

File: plugins/AudioFileProcessor/AudioFileProcessor.cpp [hunk 2]
```diff
@@ -1160,14 +1160,37 @@ void AudioFileProcessorWaveView::slideSampleByFrames( f_cnt_t _frames )
...
+		if (m_loopKnob)
+		{
+			m_loopKnob->slideBy(v, false);
+		}
+		if (m_startKnob)
+		{
+			m_startKnob->slideBy(v, false);
+		}
```
Comment (sakertooth) [inline plugins/AudioFileProcessor/AudioFileProcessor.cpp:1193]:
```suggestion
		m_endKnob->slideBy(v, false);
		m_loopKnob->slideBy(v, false);
		m_startKnob->slideBy(v, false);
```

**Review by sakertooth — COMMENTED** [review]
I believe we can simplify this a bit by removing the pointer if-checks. These pointers are assigned to valid knobs once the `AudioFileProcessorWaveView` has been created.

**Comment by szeli1** [comment]
I simplified the code as suggested. Are there any more problems?

**Review by DomClark — APPROVED** [review]

**Review by sakertooth — APPROVED** [review]

**Comment by sakertooth** [comment]
> I simplified the code as suggested. Are there any more problems?

Nope, it looks good to go :+1:.

**Pull request merged**